posthog/models/person/person.py

```diff
@@ -121,10 +121,11 @@ class PersonOverride(models.Model):
     This model has a set of constraints to ensure correctness:
     1. Unique constraint on (team_id, old_person_id) pairs.
     2. Check that old_person_id is different to override_person_id for every row.
-    3. Exclude rows that overlap across old_person_id and override_person_id (e.g. if
-        a row exists with old_person_id=123 then we would not allow a row with
+    3. Same person id cannot be used as an old_person_id and an override_person_id (per team)
```

Comment:
these changes should be in https://github.com/PostHog/posthog/pull/14477

Reply:
I've moved this to #14477 and rebased this PR.